System/DataModule.cpp

```diff
@@ -1,764 +1,497 @@
+		if (type.empty() || type == "All") {
+			map<std::string, std::list<std::pair<std::string, Entity *> > >::iterator clsItr = m_TypeMap.find("Entity");
+			if (clsItr == m_TypeMap.end()) { return false; }
+
+			// Go through only the Entity typelist since it should contain all entities in this DataModule
+			for (std::list<std::pair<std::string, Entity *> >::iterator instItr = clsItr->second.begin(); instItr != clsItr->second.end(); ++instItr) {
+				// Get the grouped entities, without transferring ownership
+				if (instItr->second->IsInGroup(group)) {
+					entityList.push_back(instItr->second);
+					foundAny = true;
+				}
+			}
+		} else {
+			// Look only in one specific class (which will get all derived classes' entities too!)
+			// Look for the class entry
+			map<std::string, std::list<std::pair<std::string, Entity *> > >::iterator clsItr = m_TypeMap.find(type);
+			if (clsItr == m_TypeMap.end()) { return 0; }
+
+			RTEAssert(!clsItr->second.empty(), "DataModule has class entry without instances in its map!?");
+
+			for (std::list<std::pair<std::string, Entity *> >::iterator instItr = clsItr->second.begin(); instItr != clsItr->second.end(); ++instItr) {
+				// Get the grouped entities, without transferring ownership
+				if (instItr->second->IsInGroup(group)) {
+					entityList.push_back(instItr->second);
+					foundAny = true;
+				}
+			}
+		}
+		return foundAny;
+	}
+
+/////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+	bool DataModule::GetAllOfType(std::list<Entity *> &entityList, std::string type) {
+		if (type.empty()) { return false; }
+
+		// Look for the class entry
+		map<std::string, std::list<std::pair<std::string, Entity *> > >::iterator clsItr = m_TypeMap.find(type);
+		if (clsItr == m_TypeMap.end()) { return false; }
+
+		RTEAssert(!clsItr->second.empty(), "DataModule has class entry without instances in its map!?");
+
+		// Found something, so let's add it to the list!
+		for (std::list<std::pair<std::string, Entity *> >::iterator instItr = clsItr->second.begin(); instItr != clsItr->second.end(); ++instItr) {
+			// Get the entities, without transferring ownership
+			entityList.push_back(instItr->second);
+		}
+		return true;
+	}
+
+/////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+	bool DataModule::AddMaterialMapping(int fromID, int toID) {
+		RTEAssert(fromID > 0 && fromID < c_PaletteEntriesNumber && toID > 0 && toID < c_PaletteEntriesNumber, "Tried to make an out-of-bounds Material mapping");
+
+		bool clear = m_MaterialMappings[fromID] == 0;
+		m_MaterialMappings[fromID] = toID;
+
+		return clear;
+	}
+
+/////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+	int DataModule::LoadScripts() {
+		if (m_ScriptPath != "") { return g_LuaMan.RunScriptFile(m_ScriptPath); }
```

Comment:
I feel like this one could be moved into the header. Also, replaced with a ternary:
```
return m_ScriptPath == "" ? 0 : g_LuaMan.RunScriptFile(m_ScriptPath);
```

Reply:
Moving it into the header made it complain that it didn't know what g_LuaMan is, which'd mean another include, so I guess it stays in the cpp where it works fine somehow.